server/src/main/java/org/opensearch/plugins/PluginsService.java

```diff
@@ -321,12 +321,19 @@ public PluginsAndModules info() {
         return info;
     }
 
+    /**
+     * Represents a plugin bundle that consists of plugin information and associated URLs.
+     * A bundle encapsulates all the resources needed for a plugin, including the plugin descriptor
+     * and the URLs for all JAR files that make up the plugin.
+     *
+     * @opensearch.internal
+     */
     // a "bundle" is a group of jars in a single classloader
-    static class Bundle {
-        final PluginInfo plugin;
+    public static class Bundle {
```

Comment:
Please bring back the package protected visibility, these classes where not designed as public

Reply:
Ya PluginsService.java the methods and class is used by by both `:server` and `:plugin-cli`, with this refactoring some of them had to be changed to public.

Reply:
You are right, getPluginBundles exposes the Bundle, but we don't need to expose constructors and `plugin`, this is for sure

Reply:
Let me explore few ways here.